### PR TITLE
update datafusion to 14 and arrow to 26

### PR DESCRIPTION
**Pull request by houqp**

### CODE REVIEW

**Comment by houqp** [comment]
@jychen7 FYI since this might cause conflict to https://github.com/roapi/roapi/pull/241

**Comment by jychen7** [comment]
nice, Datafusion 14 has better error handling for ObjectStoreProvider than 12

will rebase

File: Cargo.lock [hunk 1]
```diff
@@ -2419,19 +2795,21 @@ dependencies = [
 
 [[package]]
 name = "object_store"
-version = "0.5.0"
+version = "0.5.3"
```
Comment (jychen7) [inline Cargo.lock:2798]: dummy question: while datafusion 14 uses object_store 0.5.0, why Cargo is allow to upgrade to 0.5.3?

https://github.com/apache/arrow-datafusion/blob/14.0.0/datafusion/core/Cargo.toml#L81

Comment (houqp) [reply Cargo.lock:2798]: pinning to 0.5.0 is equivalent to pinning to 0.5.x i believe.

Comment (jychen7) [reply Cargo.lock:2798]: oh, right, I should take detail look at it (was from Ruby background and thought they are similar, turn out NOT)

https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html

```
0.2.3  :=  >=0.2.3, <0.3.0
1.2.3  :=  >=1.2.3, <2.0.0
```

**Pull request merged**